sdk-extensions/autoconfigure-spi/src/main/java/io/opentelemetry/sdk/autoconfigure/spi/ConfigProperties.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.autoconfigure.spi;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+
+/** Properties used for auto-configuration of the OpenTelemetry SDK components. */
+public interface ConfigProperties {
```

Comment:
It doesn't feel to me like this belongs in an SPI package. This interface is useful for non-SPI-based configuration as well, isn't it? Are we really thinking that we should restrict this to being SPI only?

Reply:
An alternative would be for the SPI classes to not take `ConfigProperties` as an object for any creation methods and instead have the implementation retrieve the config from `ConfigProperties` with `get()`.

`ConfigProperties` could also have an INSTANCE that is stored on "first use". I saw many usages of `ConfigProperties.get()` which essentially reloads the object and properties, but I'm not sure the benefits of the multiple creations.

Reply:
@anuraaga thoughts?